File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
  let answer = 0;
  const visit = new Array(n).fill(false);
  const queue = [];

  for (let i = 0; i < n; i++) {
    if (visit[i] === true) continue;
      
    answer++;
    visit[i] = true;
    queue.push(i);
      
    while (queue.length > 0) {
      const current = queue.shift();
        
      for (let next = 0; next < n; next++) {
        if (next === current) continue;
        if (computers[current][next] === 0) continue;
        if (visit[next] === true) continue;
          
        visit[next] = true;
        queue.push(next);
      }
    }
  }
    
  return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def dfs(cur):
        for next in range(len(computers)):
            if not computers[cur][next]:
                continue
            if visited[next]:
                continue
                
            visited[next] = True
            dfs(next)
    
    for i in range(n):
        if visited[i]:
            continue
            
        visited[i] = True
        dfs(i)
        answer += 1
            
    return answer
